miasm2/arch/aarch64/arch.py

```diff
@@ -0,0 +1,1968 @@
+            return str(e)
+        elif isinstance(e, m2_expr.ExprOp) and e.op in shift_expr:
+            op_str = shift_str[shift_expr.index(e.op)]
+            return "%s %s %s" % (e.args[0], op_str, e.args[1])
+        elif isinstance(e, m2_expr.ExprOp) and e.op == "slice_at":
+            return "%s LSL %s" % (e.args[0], e.args[1])
+        elif isinstance(e, m2_expr.ExprOp) and e.op in extend_lst:
+            op_str = e.op
+            return "%s %s %s" % (e.args[0], op_str, e.args[1])
+        elif isinstance(e, m2_expr.ExprOp) and e.op == "postinc":
+            if e.args[1].arg != 0:
+                return "[%s], %s" % (e.args[0], e.args[1])
+            else:
+                return "[%s]" % (e.args[0])
+        elif isinstance(e, m2_expr.ExprOp) and e.op == "preinc_wb":
+            if e.args[1].arg != 0:
+                return "[%s, %s]!" % (e.args[0], e.args[1])
+            else:
+                return "[%s]" % (e.args[0])
+        elif isinstance(e, m2_expr.ExprOp) and e.op == "preinc":
+            if len(e.args) == 1:
+                return "[%s]" % (e.args[0])
+            elif not isinstance(e.args[1], m2_expr.ExprInt) or e.args[1].arg != 0:
+                return "[%s, %s]" % (e.args[0], e.args[1])
+            else:
+                return "[%s]" % (e.args[0])
+        elif isinstance(e, m2_expr.ExprOp) and e.op == 'segm':
+            arg = e.args[1]
+            if isinstance(arg, m2_expr.ExprId):
+                arg = str(arg)
+            elif arg.op == 'LSL' and arg.args[1].arg == 0:
+                arg = str(arg.args[0])
+            else:
+                arg = "%s %s %s" % (arg.args[0], arg.op, arg.args[1])
+            return '[%s, %s]' % (e.args[0], arg)
+
+        fds
+        return o
+
+    def dstflow(self):
+        return self.name in self.name in brcond + ["B", "BL"]
+
+    def mnemo_flow_to_dst_index(self, name):
+        if self.name in ['CBZ', 'CBNZ']:
+            return 1
+        elif self.name in ['TBZ', 'TBNZ']:
+            return 2
+        else:
+            return 0
+
+    def dstflow2label(self, symbol_pool):
+        index = self.mnemo_flow_to_dst_index(self.name)
+        e = self.args[index]
+        if not isinstance(e, m2_expr.ExprInt):
+            return
+        ad = e.arg + self.offset
+        l = symbol_pool.getby_offset_create(ad)
+        s = m2_expr.ExprId(l, e.size)
+        self.args[index] = s
+
+    def breakflow(self):
+        if self.name in brcond + ["BR", "BLR", "RET", "ERET", "DRPS", "B", "BL"]:
+            return True
+        return False
+
+    def is_subcall(self):
+        if self.name in ["BLR", "BL"]:
+            return True
+        return False
+
+    def getdstflow(self, symbol_pool):
+        index = self.mnemo_flow_to_dst_index(self.name)
+        return [self.args[index]]
+
+    def splitflow(self):
+        if self.name in brcond + ["BLR", "BL"]:
+            return True
+        return False
+
+    def get_symbol_size(self, symbol, symbol_pool):
```

Comment:
This comment is not related only to this PR. Some of the elements of this class look like `property`, as `symbol_size`. May for a future arch refactoring